pennylane_forest/qc.py

```diff
@@ -0,0 +1,296 @@
+"""
+Base Quantum Computer device class
+==================================
+
+**Module name:** :mod:`pennylane_forest.qc`
+
+.. currentmodule:: pennylane_forest.qc
+
+This module contains the :class:`~.QuantumComputerDevice` base class that can be
+used to build PennyLane devices from pyQuil QuantumComputers (such as a Rigetti Quantum Processor Unit).
+
+Classes
+-------
+
+.. autosummary::
+   QuantumComputerDevice
+
+Code details
+~~~~~~~~~~~~
+"""
+
+from abc import ABC, abstractmethod
+from typing import Dict
+from collections import OrderedDict
+
+from pyquil import Program
+from pyquil.api import QAMExecutionResult, QuantumComputer
+from pyquil.gates import RESET, MEASURE
+from pyquil.quil import Pragma
+
+from pennylane import DeviceError, numpy as np
+from pennylane.wires import Wires
+
+from .device import ForestDevice
+from ._version import __version__
+
+
+class QuantumComputerDevice(ForestDevice, ABC):
+    r"""Abstract Quantum Computer device for PennyLane.
+
+    This is a base class for common logic shared by pyQuil ``QuantumComputer``s (i.e. QVMs, QPUs).
+    Children classes need to define a `get_qc` method that returns a pyQuil ``QuantumComputer``.
+
+    Args:
+        device (str): the name of the device to initialise.
+        shots (int): number of circuit evaluations/random samples used
+            to estimate expectation values of observables.
+         wires (Iterable[Number, str]): Iterable that contains unique labels for the
+            qubits as numbers or strings (i.e., ``['q1', ..., 'qN']``).
+            The number of labels must match the number of qubits accessible on the backend.
+            If not provided, qubits are addressed as consecutive integers ``[0, 1, ...]``, and their number
+            is inferred from the backend.
+        active_reset (bool): whether to actively reset qubits instead of waiting for
+            for qubits to decay to the ground state naturally.
+            Setting this to ``True`` results in a significantly faster expectation value
+            evaluation when the number of shots is larger than ~1000.
+
+    Keyword args:
+        compiler_timeout (int): number of seconds to wait for a response from quilc (default 10).
+        execution_timeout (int): number of seconds to wait for a response from the QVM (default 10).
+        parametric_compilation (bool): a boolean value of whether or not to use parametric
+            compilation.
+    """
+    pennylane_requires = ">=0.17"
```

Comment:
```suggestion
    pennylane_requires = ">=0.18"
```

Reply:
Set the min version of pennylane to `0.18` everywhere.